tapdb/universe.go

```diff
@@ -614,6 +613,62 @@ func universeFetchProofLeaf(ctx context.Context,
 	return proofs, nil
 }
 
+// mintingKeys returns all the leaf keys in the target universe.
+func mintingKeys(ctx context.Context, dbTx BaseUniverseStore,
+	q universe.UniverseLeafKeysQuery,
+	namespace string) ([]universe.LeafKey, error) {
+
+	universeKeys, err := dbTx.FetchUniverseKeys(
+		ctx, UniverseLeafKeysQuery{
+			Namespace:     namespace,
+			SortDirection: sqlInt16(q.SortDirection),
+			NumOffset:     q.Offset,
+			NumLimit: func() int32 {
+				if q.Limit == 0 {
+					return universe.MaxPageSize
+				}
+
+				return q.Limit
+			}(),
+		},
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	var leafKeys []universe.LeafKey
```

Comment:
maybe a place we'd want iterators over preallocated slices in addition to what we have now.